src/api/helpers/OperationValidator.ts

```diff
@@ -1,25 +1,24 @@
 import { arrayBufferToBase64 } from "@/use/crypto/certificates";
-import Operation from "../api_models/operation_management/Operation";
+import Operation, { TransactionInfo } from "../api_models/operation_management/Operation";
 
 export async function validateOperationId(operation: Operation) {
+    return (await calculateOperationId(operation.transactionInfo)) === operation.operationId;
+}
+
+export async function calculateOperationId(transactionInfo: TransactionInfo) {
     const crypto = window.crypto.subtle;
 
     const toHash =
-        operation.transactionInfo.contractName +
-        ":" +
-        operation.transactionInfo.transactionName +
-        ":" +
-        operation.transactionInfo.parameters.replace(/\s/g, "");
+        transactionInfo.contractName + ":" + transactionInfo.transactionName + ":" + transactionInfo.parameters.replace(/\s/g, "");
 
     const operationHash = await crypto.digest("SHA-256", new Uint8Array(toUTF8Array(toHash)));
 
     let operationId = arrayBufferToBase64(operationHash);
-
-    //url base64
     operationId = operationId.replace(/\+/g, "-");
     operationId = operationId.replace(/\//g, "_");
+    //operationId = operationId.replace(/=/g, "");
```

Comment:
```suggestion
```

Reply:
We will need this with the next chaincode version, I am not sure if this branch will be merged before we test that version.